src/Agent.Worker/TestResults/TestRunPublisher.cs

```diff
@@ -154,6 +160,34 @@ public TestRunData ReadResultsFromFile(TestRunContext runContext, string filePat
         }
         #endregion
 
+        private bool HierarchicalSubresultPreProcessing(string automatedTestName, List<TestCaseSubResultData> subResults, int level = 1)
```

Comment:
the method name doesn't inform it will return a Bool, do you mind change to a more readable name? :D